### PR TITLE
chore: migrate to yarn 2

### PR DESCRIPTION
**Pull request by lazpavel**
#### NOTE
Test changes will be added as a follow-up PR. All yarn migration PRs will point to yarn-migration-main branch and only merged to dev once all the work is completed.

[Pipeline Link](https://app.circleci.com/pipelines/github/aws-amplify/amplify-cli/19581/workflows/1d9a452f-a0e8-4ccc-97be-bb806813977a): This PR will only pass the packages build part, the build tests will still fail and be addressed in the followup PR

#### Description of changes
- changes package.json to use Yarn 2
- updates package.json scripts to Yarn 2 syntax
- reverts the use of Nx back to Lerna as it now delegates to Nx internally
- updates jest to version 29
- updates Lerna to 6.6.1

#### Description of how you validated changes
- manual build tested on macOS, Linux, and Windows
- build passes on circle CI (not the test steps)
 
#### Checklist
- [x] PR description included

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by danielleadams** [comment]
@lazpavel is this ready for review?

**Comment by lazpavel** [comment]
@danielleadams, want to fix windows build as part of this PR as well, will set it back to ready for review after I test that

File: package.json [hunk 1]
```diff
@@ -59,12 +58,11 @@
     "verdaccio-disconnect": "source .circleci/local_publish_helpers.sh && unsetNpmRegistryUrl",
     "verdaccio-start": "source .circleci/local_publish_helpers.sh && startLocalRegistry \"$(pwd)/.circleci/verdaccio.yaml\"",
     "verdaccio-stop": "kill -9 $(lsof -n -t -iTCP:4873 -sTCP:LISTEN) || true",
-    "verify-api-extract:clean": "yarn clean && yarn dev-build && yarn verify-api-extract",
+    "verify-api-extract:clean": "yarn clean && yarn build && yarn verify-api-extract",
     "verify-api-extract": "yarn extract-api && ./scripts/verify-extract-api.sh",
     "verify-commit": "yarn ts-node ./scripts/verify-commit.ts",
     "verify-yarn-lock": "./scripts/verify-yarn-lock.sh",
-    "workflow-results": "yarn ts-node ./scripts/cci-workflow-results.ts",
-    "yarn-use-bash": "yarn config set script-shell /bin/bash"
```
Comment (lazpavel) [inline package.json:67]: yarn no longer supports script shell and is using an internal shell interpreter: https://github.com/yarnpkg/berry/discussions/2589

File: package.json [hunk 2]
```diff
@@ -4,26 +4,25 @@
...
+    "clean": "lerna run clean && lerna exec yarn rimraf tsconfig.tsbuildinfo && lerna clean --yes && yarn rimraf node_modules",
     "cloud-e2e-with-rc": "CURR_BRANCH=$(git branch | awk '/\\*/{printf \"%s\", $2}') && UPSTREAM_BRANCH=run-e2e-with-rc/$USER/$CURR_BRANCH && git push $(git remote -v | grep aws-amplify/amplify-cli | head -n1 | awk '{print $1;}') $CURR_BRANCH:$UPSTREAM_BRANCH --no-verify --force-with-lease && echo \"\n\n 🏃 E2E test are running at:\nhttps://app.circleci.com/pipelines/github/aws-amplify/amplify-cli?branch=$UPSTREAM_BRANCH\"",
     "cloud-e2e": "CURR_BRANCH=$(git branch | awk '/\\*/{printf \"%s\", $2}') && UPSTREAM_BRANCH=run-e2e/$USER/$CURR_BRANCH && git push $(git remote -v | grep aws-amplify/amplify-cli | head -n1 | awk '{print $1;}') $CURR_BRANCH:$UPSTREAM_BRANCH --no-verify --force-with-lease && echo \"\n\n 🏃 E2E test are running at:\nhttps://app.circleci.com/pipelines/github/aws-amplify/amplify-cli?branch=$UPSTREAM_BRANCH\"",
     "commit": "git-cz",
     "coverage:collect": "ts-node ./scripts/collect-test-coverage.ts",
     "coverage": "codecov || exit 0",
-    "dev-build": "yarn --cache-folder ~/.cache/yarn && nx run-many --target=build --all",
```
Comment (lazpavel) [inline package.json:19]: yarn no longer supports --cache-folder argument, and is now configured in yarnrc.yml file, this makes `dev-build` and `build` scripts redundant

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -37,13 +49,13 @@ This section should get you running with **Amplify CLI** and get you familiar wi
 
    > If you are using Yarn v2, run `yarn set version classic` to change to Yarn Classic.
 
-   > Ensure that [Yarn global bin](https://classic.yarnpkg.com/en/docs/cli/global) is added to your PATH. For example, add `export PATH="$(yarn global bin):$PATH"` to your shell profile file on Linux or macOS.
+   > Ensure that `.bin` directory is added to your PATH. For example, add `export PATH="<amplify-cli/.bin>:$PATH"` to your shell profile file on Linux or macOS.
```
Comment (danielleadams) [inline CONTRIBUTING.md:52]: Should this be added the one of the setup scripts?

**Review by danielleadams — COMMENTED** [review]
@lazpavel I started reviewing - just wanted to leave a few initial comments and I'll finish up tomorrow.

File: .circleci/config.yml [hunk 4]
```diff
@@ -52,6 +52,7 @@ jobs:
           name: Generate config
           command: |
             cd scripts
+            yarn --version
```
[the remark this answers is not in the document]
Comment (lazpavel) [reply .circleci/config.yml:55]: not anymore, added for testing and forgot to remove

File: CONTRIBUTING.md (hunk 3 above)
Comment (lazpavel) [reply CONTRIBUTING.md:52]: we did not used to do it before, and in my opinion the setup script should not mess with local env variables, but I don't have strong opinions about this

File: .circleci/config.base.yml [hunk 5]
```diff
@@ -133,9 +133,8 @@ jobs:
     executor: l_xlarge
     steps:
       - checkout
-      - run: yarn config set registry https://registry.npmjs.org
-      - run: yarn config set script-shell $(which bash)
-      - run: yarn install --network-concurrency 1
+      - run: yarn config set npmRegistryServer https://registry.npmjs.org
+      - run: yarn install
```
[the remark this answers is not in the document]
Comment (lazpavel) [reply .circleci/config.base.yml:137]: added

File: .circleci/config.base.yml [hunk 6]
```diff
@@ -749,7 +748,7 @@ jobs:
           name: Configure Amplify CLI
           command: |
             yarn rm-dev-link && yarn link-dev && yarn rm-aa-dev-link && yarn link-aa-dev
-            echo 'export PATH="$(yarn global bin):$PATH"' >> $BASH_ENV
+            echo 'export PATH="$(pwd)/.bin:$PATH"' >> $BASH_ENV
```
[the remark this answers is not in the document]
Comment (lazpavel) [reply .circleci/config.base.yml:751]: yarn no longer has a global bin directory; I changed the setup scripts to create a .bin directory in <rootDir> in order to copy the amplify-dev binary.

File: CONTRIBUTING.md (hunk 3 above)
Comment (evcodes) [reply CONTRIBUTING.md:52]: I would add it to the setup scripts to remove an extra step that contributors have to do in order to setup their local environment

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -37,13 +49,13 @@ This section should get you running with **Amplify CLI** and get you familiar wi
...
 
-1. Ensure you have [Java](https://aws.amazon.com/corretto/) installed and `java` command is available in your system. This is required for DynamoDB emulator.
+2. Ensure you have [Java](https://aws.amazon.com/corretto/) installed and `java` command is available in your system. This is required for DynamoDB emulator.
 
-1. Ensure you are using the npm registry, even with yarn by running `yarn config set registry https://registry.npmjs.org`
+3. Ensure you are using the npm registry, even with yarn by running `yarn config set npmRegistryServer https://registry.npmjs.org`
 
-1. Start by [forking](https://docs.github.com/en/github/getting-started-with-github/fork-a-repo) the _dev_ branch of [amplify-cli](https://github.com/aws-amplify/amplify-cli). Then clone it to your machine to work with it locally using one of the following methods:
+4. Start by [forking](https://docs.github.com/en/github/getting-started-with-github/fork-a-repo) the _dev_ branch of [amplify-cli](https://github.com/aws-amplify/amplify-cli). Then clone it to your machine to work with it locally using one of the following methods:
```
Comment (evcodes) [inline CONTRIBUTING.md:58]: Good updates on this contributing doc.

File: package.json [hunk 8]
```diff
@@ -4,26 +4,25 @@
...
-    "build-tests": "nx run-many --target=build-tests --all",
-    "build": "nx run-many --target=build --all",
+    "build-and-verify": "yarn clean && yarn build && yarn prettier-check && yarn lint-check && yarn verify-api-extract",
+    "build-tests-changed": "lerna run build-tests --since dev",
+    "build-tests": "lerna run build-tests",
+    "build": "lerna run build",
     "check-version-conflicts:beta": "yarn ts-node ./scripts/check-version.ts beta",
     "check-version-conflicts:release": "yarn ts-node ./scripts/check-version.ts",
-    "clean": "nx reset && nx run-many --target=clean --all && yarn rimraf tsconfig.tsbuildinfo",
+    "clean": "lerna run clean && lerna exec yarn rimraf tsconfig.tsbuildinfo && lerna clean --yes && yarn rimraf node_modules",
```
Comment (evcodes) [inline package.json:13]: When I ran yarn clean, I got a Usage Error saying that it can't find a script named "rimraf"

Comment (lazpavel) [reply package.json:13]: can you try running `yarn && yarn clean`, it looks that it cannot find the dependency that might need to be installed first

File: .yarnrc.yml [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+compressionLevel: 0
+
+enableGlobalCache: true
+
+logFilters:
+  - code: YN0060
+    level: discard
+  - code: YN0002
+    level: discard
+  - code: YN0007
```
[the remark this answers is not in the document]
Comment (lazpavel) [reply .yarnrc.yml:12]: for example: https://yarnpkg.com/advanced/error-codes#yn0002---missing_peer_dependency

**Review by evcodes — APPROVED** [review]

File: package.json (hunk 8 above)
Comment (akshbhu) [reply package.json:13]: I think its taking global rimraf for eddy. Should this be: `lerna exec yarn ./node_modules/bin/rimraf  tsconfig.tsbuildinfo`

File: package.json [hunk 10]
```diff
@@ -4,26 +4,25 @@
...
     "coverage": "codecov || exit 0",
-    "dev-build": "yarn --cache-folder ~/.cache/yarn && nx run-many --target=build --all",
-    "e2e": "nx run e2e",
-    "extract-api": "nx run-many --target=extract-api --all",
+    "e2e": "lerna run e2e",
+    "extract-api": "lerna run extract-api",
     "extract-formatting-changes": "yarn ts-node ./scripts/extract-formatting-changes.ts",
     "finish-release": "ts-node ./scripts/finish-release.ts",
-    "link-aa-dev": "cd packages/amplify-app && ln -s \"$(pwd)/bin/amplify-app\" \"$(yarn global bin)/amplify-app-dev\" && cd -",
+    "link-aa-dev": "mkdir -p .bin/ && cd packages/amplify-app && ln -s \"$(pwd)/bin/amplify-app\" ../../.bin/amplify-app-dev && cd ../../",
```
Comment (akshbhu) [inline package.json:23]: `cd ../../` , why is this needed at the end?

File: packages/amplify-container-hosting/package.json [hunk 11]
```diff
@@ -33,11 +33,19 @@
     "mime-types": "^2.1.26",
     "promise-sequential": "^1.1.1"
   },
+  "peerDependencies": {
```
Comment (akshbhu) [inline packages/amplify-container-hosting/package.json:36]: why this is added as peer dependency?

File: packages/amplify-graphiql-explorer/package.json [hunk 12]
```diff
@@ -29,7 +29,7 @@
     "eslint-webpack-plugin": "^3.1.1",
     "file-loader": "^6.2.0",
     "fs-extra": "^10.0.0",
-    "graphiql": "^1.3.2",
+    "graphiql": "1.5.16",
```
Comment (akshbhu) [inline packages/amplify-graphiql-explorer/package.json:32]:
```suggestion
    "graphiql": "^1.5.16", ?
```

File: package.json (hunk 10 above)
Comment (lazpavel) [reply package.json:23]: used to be `&& cd -` but this was failing after I added the `mkdir`

File: packages/amplify-container-hosting/package.json (hunk 11 above)
Comment (lazpavel) [reply packages/amplify-container-hosting/package.json:36]: it should not, will update

File: packages/amplify-graphiql-explorer/package.json (hunk 12 above)
Comment (lazpavel) [reply packages/amplify-graphiql-explorer/package.json:32]: is question mark a typo? otherwise, I believe we can make this change

File: package.json (hunk 8 above)
Comment (lazpavel) [reply package.json:13]: changing it to just `yarn rimraf` without lerna exec seems to fix it, will update

File: packages/amplify-appsync-simulator/package.json [hunk 13]
```diff
@@ -66,6 +66,13 @@
     "@types/ws": "^8.2.2",
     "jsonwebtoken": "^9.0.0"
   },
+  "packageExtensions": {
+    "graphql-iso-date": {
+      "peerDependencies": {
+        "graphql": "^15.8.0"
```
Comment (akshbhu) [inline packages/amplify-appsync-simulator/package.json:72]: why is this added ? , 
graphql is present in the dependencies

File: .yarnrc.yml [hunk 14]
```diff
@@ -0,0 +1,25 @@
+cacheFolder: ~/.yarn/berry/cache
+
+compressionLevel: 0
+
+enableGlobalCache: true
```
Comment (akshbhu) [inline .yarnrc.yml:5]: Do we need to set global folder here if this flag is true : 
https://yarnpkg.com/configuration/yarnrc#global

**Review by akshbhu — APPROVED** [review]
Just few questions for my knowledge else LGTM!!
